src/image_provider/reqwest.rs

```diff
@@ -82,17 +83,45 @@ pub mod client {
                 })
                 .collect();
             if status.is_success() {
-                let bytes = response.bytes().await.map_err(|e| {
-                    error!(
-                        "failed to read the binary payload of the image '{}'. error: {}",
-                        resource, e
-                    );
-                    ImageDownloadFailed
-                })?;
-                Ok(ImageResponse {
-                    bytes: bytes.to_vec(),
-                    response_headers: headers,
-                })
+                if let Some(max_size) = config.max_file_size {
```

Comment:
This part isn't ok. We must continue to support downloading the entire file if the `max_file_size` isn't specified in the configuration. That's for backward compatibility purposes. We have to consider the people that currently use Dali and will want to upgrade to the newer image.

Reply:
No, it's fine to keep it in chunks for reqwest and only check if the the config param is set.